src/ui/pages/CreatePassword/CreatePassword.tsx

```diff
@@ -6,59 +6,94 @@ import { useAppDispatch, useAppSelector } from "../../../store/hooks";
 import {
   getStateCache,
   setCurrentOperation,
+  setToastMsg,
 } from "../../../store/reducers/stateCache";
 import { updateReduxState } from "../../../store/utils";
 import { PageHeader } from "../../components/PageHeader";
 import { PasswordModule } from "../../components/PasswordModule";
 import { ScrollablePageLayout } from "../../components/layout/ScrollablePageLayout";
-import { OperationType } from "../../globals/types";
+import { OperationType, ToastMsgType } from "../../globals/types";
 import { useAppIonRouter } from "../../hooks";
 import "./CreatePassword.scss";
 import { PasswordModuleRef } from "../../components/PasswordModule/PasswordModule.types";
+import { CreatePasswordProps } from "./CreatePassword.types";
 
-const CreatePassword = () => {
+const CreatePassword = ({
+  isModal,
+  handleClear,
+  setPasswordIsSet,
+  userAction,
+}: CreatePasswordProps) => {
   const pageId = "create-password";
+  const dispatch = useAppDispatch();
   const stateCache = useAppSelector(getStateCache);
   const ionRouter = useAppIonRouter();
-  const dispatch = useAppDispatch();
-
   const passwordModuleRef = useRef<PasswordModuleRef>(null);
 
   const handleContinue = async (skipped: boolean) => {
-    const { nextPath, updateRedux } = getNextRoute(RoutePath.CREATE_PASSWORD, {
-      store: { stateCache },
-      state: { skipped },
-    });
+    if (isModal) {
```

Comment:
Maybe a personal thing but to me if it's not a model on onboarding and is everywhere else - a variable like `isOnboarding` or something would be clearer to know what's happening when seeing this for the first time